### PR TITLE
Made tests fault tolerant to missing crypten module for windows

### LINKED ISSUE #4025: Check dependency CrypTen tests
## Description
Currently, CrypTen is available only on Linux and Mac, but if someone installs the PySyft repo and run the tests we also have some CrypTen related tests.
Those should be taken into consideration only if we have ```dependency_check.crypten```.

## How to Reproduce
Run the tests on Windows

## Expected Behavior
All the tests should pass - maybe skip the tests for CrypTen.

### PR DESCRIPTION
**Pull request by Karnak123**
## Description
Resolves #4025 by skipping crypten test files and adding expected to fail in serde tests if operating system is Windows.

## How has this been tested?
My own system was failing some tests and `test_inv_sym` goes on forever, so I ran only crypten and serde tests by temporarily moving other test files to another destination.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by Karnak123** [comment]
The fail shows `FAILED test/torch/tensors/test_precision.py::test_torch_sigmoid_approx[exp-3-0.065]` and I didn't touch the function and neither is it dependent on anything I changes/added, so do I need to worry about it?

**Comment by gmuraru** [comment]
Can't we have a skip method for the entire file (rather than marking each individual test?)

**Comment by Karnak123** [comment]
Not any way I know of, if they were all in a class or module we could have with pytest.mark.skipif(), to skip entire file only pytest.mark.skip is available but this unconditionally skips all tests, so the tests wouldn't run on Linux and Mac too

**Comment by gmuraru** [comment]
> Not any way I know of, if they were all in a class or module we could have with pytest.mark.skipif(), to skip entire file only pytest.mark.skip is available but this unconditionally skips all tests, so the tests wouldn't run on Linux and Mac too

It seems that we can skip a file using the ```conftest.py```. There are more details [here](https://docs.pytest.org/en/latest/example/pythoncollection.html#customizing-test-collection) - did not test it, but looks promising

**Comment by Karnak123** [comment]
Yeah that worked thanks! So now I'm checking if its a Windows system and ignoring all test in `test\crypten`, but still using individual checks for the other two files because only a small part of their checks depend on crypten, should I skip them as files them too or let them be? Currently they are marked as expected to fail

**Comment by Syzygianinfern0** [comment]
> should I skip them as files them too or let them be? Currently they are marked as expected to fail

I guess having it as an xfail is sufficient.

File: test/serde/msgpack/test_msgpack_serde_full.py [hunk 1]
```diff
@@ -98,6 +100,7 @@
 samples[SerializableDummyClass] = make_serializable_dummy_class
 
 
+@pytest.mark.xfail("os_name == 'nt'")
```
Comment (Syzygianinfern0) [inline test/serde/msgpack/test_msgpack_serde_full.py:103]: Could you add the `reason` attribute to these?

**Review by Syzygianinfern0 — CHANGES_REQUESTED** [review]
We can have it merged right after that! :tada:

File: test/serde/msgpack/test_msgpack_serde_full.py (hunk 1 above)
Comment (Karnak123) [reply test/serde/msgpack/test_msgpack_serde_full.py]: Of course, added attribute

**Review by Syzygianinfern0 — APPROVED** [review]

**Comment by gmuraru** [comment]
There is also a coverage issue - You might need to ignore the crypten related files for coverage if windows is detected

**Comment by Karnak123** [comment]
> There is also a coverage issue - You might need to ignore the crypten related files for coverage if windows is detected

Looked into it, couldn't find any such issues in the tests

**Comment by iamtrask** [comment]
OpenCI seems to be hanging?

**Comment by Syzygianinfern0** [comment]
> OpenCI seems to be hanging?

Yep

**Comment by Karnak123** [comment]
Excuse me? Didn't understand

**Comment by Karnak123** [comment]
Hi, WIP check hasn't still completed, so should I re-commit the code? Or is there anything else I need to change?

**Comment by gmuraru** [comment]
Hmm..Restarted the tests

**Comment by Karnak123** [comment]
I pushed an empty commit, and the test fail is due to 94% test coverage but in the last commit it was passing and I haven't made any changes since so not sure what to do :/

**Comment by gmuraru** [comment]
Ahh...I do not know how this got out of my notification bar from GitHub.
Could you reopen it, please? - I can check it out.

**Pull request closed without merging**